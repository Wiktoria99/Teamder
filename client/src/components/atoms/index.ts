export * from './BackButton';
export * from './CustomTextField';
export * from './CustomButton';
export * from './SideBarButton';
