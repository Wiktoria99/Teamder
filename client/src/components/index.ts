export * from "./atoms";
export * from "./atoms";
export * from "./molecules";
export * from "./organisms";
