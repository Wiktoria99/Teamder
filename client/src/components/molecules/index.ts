export * from './RegisterSteps';
export * from './TeamItem';
export * from './Suggestions';
export * from './Sidebar';
export * from './TeamList';
export * from './SuggestionItem';
export * from './SuggestionList';
export * from './NotificationItem';
export * from './NotificationList';
