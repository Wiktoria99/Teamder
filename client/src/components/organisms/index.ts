export * from './Layout';
export * from './FormWrapper';
export * from './MainWrapper';
export * from './BarWrapper';
