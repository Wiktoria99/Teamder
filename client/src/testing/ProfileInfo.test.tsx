import React from 'react';
import ReactDOM from 'react-dom';
// import {BrowserRouter as Router, NavLink} from 'react-router-dom';
// import { StaticRouter } from 'react-router'
// import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { ProfileInfo, EditProfile, SocialMedia, UserInterests } from '@/components';


it('EditInfo renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<EditProfile /> , div);
});


it('SocialMedia renders without crashing', () => {
  const links= {
    first: '',
    second: '',
    third: '',
  };
  const div = document.createElement('div');
  ReactDOM.render(<SocialMedia links={links}/> , div);
});

it('SocialMedia matches snapshot', () => {
  const links= {
    first: '',
    second: '',
    third: '',
  };
  const tree = renderer.create(<SocialMedia links={links}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('UserInterests renders without crashing', () => {
  const interests= [''];
  const div = document.createElement('div');
  ReactDOM.render(<UserInterests interests={interests}/> , div);
});

it('UserInterests matches snapshot', () => {
  const interests= [''];
  const tree = renderer.create(<UserInterests interests={interests}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ProfileInfo renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ProfileInfo /> , div);
});