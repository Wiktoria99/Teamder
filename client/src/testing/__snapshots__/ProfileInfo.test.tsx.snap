// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMedia matches snapshot 1`] = `
Array [
  <ul
    className="MuiList-root makeStyles-root-8 MuiList-padding"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          fill="none"
          height="30"
          viewBox="0 0 30 30"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clipPath="url(#clip0)"
          >
            <path
              d="M29.9998 15.0928C29.9998 6.75652 23.2836 -0.000976562 14.9998 -0.000976562C6.71234 0.000898438 -0.00390625 6.75652 -0.00390625 15.0946C-0.00390625 22.6265 5.48234 28.8703 12.6523 30.0028V19.4559H8.84609V15.0946H12.6561V11.7665C12.6561 7.98465 14.8967 5.8959 18.3223 5.8959C19.9648 5.8959 21.6805 6.19027 21.6805 6.19027V9.90277H19.7886C17.9267 9.90277 17.3455 11.0671 17.3455 12.2615V15.0928H21.5042L20.8405 19.454H17.3436V30.0009C24.5136 28.8684 29.9998 22.6246 29.9998 15.0928Z"
              fill="#F2F2F2"
            />
          </g>
          <defs>
            <clipPath
              id="clip0"
            >
              <rect
                fill="white"
                height="30"
                width="30"
              />
            </clipPath>
          </defs>
        </svg>
      </div>
      <div
        className="MuiListItemText-root makeStyles-linkFont-11"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          fill="none"
          height="30"
          viewBox="0 0 30 30"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 0C10.9294 0 10.4175 0.01875 8.81812 0.09C7.21875 0.165 6.12937 0.41625 5.175 0.7875C4.17393 1.16404 3.26717 1.75467 2.51813 2.51813C1.75514 3.26756 1.16458 4.1742 0.7875 5.175C0.41625 6.1275 0.163125 7.21875 0.09 8.8125C0.01875 10.4156 0 10.9256 0 15.0019C0 19.0744 0.01875 19.5844 0.09 21.1838C0.165 22.7812 0.41625 23.8706 0.7875 24.825C1.17188 25.8112 1.68375 26.6475 2.51813 27.4819C3.35063 28.3162 4.18687 28.83 5.17313 29.2125C6.12937 29.5837 7.21688 29.8369 8.81437 29.91C10.4156 29.9812 10.9256 30 15 30C19.0744 30 19.5825 29.9812 21.1838 29.91C22.7794 29.835 23.8725 29.5837 24.8269 29.2125C25.8273 28.8358 26.7334 28.2451 27.4819 27.4819C28.3162 26.6475 28.8281 25.8112 29.2125 24.825C29.5819 23.8706 29.835 22.7812 29.91 21.1838C29.9812 19.5844 30 19.0744 30 15C30 10.9256 29.9812 10.4156 29.91 8.81437C29.835 7.21875 29.5819 6.1275 29.2125 5.175C28.8355 4.17417 28.2449 3.26752 27.4819 2.51813C26.7331 1.75439 25.8262 1.16371 24.825 0.7875C23.8687 0.41625 22.7775 0.163125 21.1819 0.09C19.5806 0.01875 19.0725 0 14.9963 0H15.0019H15ZM13.6556 2.70375H15.0019C19.0069 2.70375 19.4812 2.71688 21.0619 2.79C22.5244 2.85563 23.3194 3.10125 23.8481 3.30562C24.5475 3.5775 25.0481 3.90375 25.5731 4.42875C26.0981 4.95375 26.4225 5.4525 26.6944 6.15375C26.9006 6.68062 27.1444 7.47563 27.21 8.93813C27.2831 10.5188 27.2981 10.9931 27.2981 14.9963C27.2981 18.9994 27.2831 19.4756 27.21 21.0562C27.1444 22.5187 26.8988 23.3119 26.6944 23.8406C26.4539 24.4919 26.07 25.0808 25.5712 25.5637C25.0462 26.0887 24.5475 26.4131 23.8463 26.685C23.3213 26.8913 22.5262 27.135 21.0619 27.2025C19.4812 27.2738 19.0069 27.2906 15.0019 27.2906C10.9969 27.2906 10.5206 27.2738 8.94 27.2025C7.4775 27.135 6.68437 26.8913 6.15562 26.685C5.50406 26.4449 4.91461 26.0617 4.43062 25.5637C3.93141 25.0801 3.54698 24.4906 3.30562 23.8387C3.10125 23.3119 2.85563 22.5169 2.79 21.0544C2.71875 19.4738 2.70375 18.9994 2.70375 14.9925C2.70375 10.9875 2.71875 10.515 2.79 8.93437C2.8575 7.47187 3.10125 6.67688 3.3075 6.14813C3.57938 5.44875 3.90562 4.94812 4.43062 4.42312C4.95563 3.89812 5.45437 3.57375 6.15562 3.30188C6.68437 3.09563 7.4775 2.85188 8.94 2.78437C10.3238 2.72062 10.86 2.70188 13.6556 2.7V2.70375ZM23.0081 5.19375C22.7717 5.19375 22.5377 5.24031 22.3193 5.33077C22.1009 5.42123 21.9025 5.55381 21.7353 5.72096C21.5682 5.8881 21.4356 6.08653 21.3451 6.30492C21.2547 6.52331 21.2081 6.75737 21.2081 6.99375C21.2081 7.23013 21.2547 7.46419 21.3451 7.68258C21.4356 7.90097 21.5682 8.0994 21.7353 8.26654C21.9025 8.43369 22.1009 8.56627 22.3193 8.65673C22.5377 8.74719 22.7717 8.79375 23.0081 8.79375C23.4855 8.79375 23.9434 8.60411 24.2809 8.26654C24.6185 7.92898 24.8081 7.47114 24.8081 6.99375C24.8081 6.51636 24.6185 6.05852 24.2809 5.72096C23.9434 5.38339 23.4855 5.19375 23.0081 5.19375ZM15.0019 7.2975C13.9801 7.28156 12.9654 7.46904 12.0168 7.84901C11.0682 8.22899 10.2047 8.79388 9.47653 9.51079C8.74835 10.2277 8.17006 11.0823 7.77533 12.0249C7.3806 12.9674 7.17731 13.9791 7.17731 15.0009C7.17731 16.0228 7.3806 17.0345 7.77533 17.977C8.17006 18.9196 8.74835 19.7742 9.47653 20.4911C10.2047 21.208 11.0682 21.7729 12.0168 22.1529C12.9654 22.5328 13.9801 22.7203 15.0019 22.7044C17.0241 22.6728 18.9529 21.8473 20.3718 20.4061C21.7908 18.9649 22.5861 17.0234 22.5861 15.0009C22.5861 12.9784 21.7908 11.037 20.3718 9.59578C18.9529 8.15454 17.0241 7.32905 15.0019 7.2975ZM15.0019 9.99938C16.3281 9.99938 17.6001 10.5262 18.5379 11.464C19.4757 12.4018 20.0025 13.6738 20.0025 15C20.0025 16.3262 19.4757 17.5982 18.5379 18.536C17.6001 19.4738 16.3281 20.0006 15.0019 20.0006C13.6756 20.0006 12.4037 19.4738 11.4659 18.536C10.5281 17.5982 10.0013 16.3262 10.0013 15C10.0013 13.6738 10.5281 12.4018 11.4659 11.464C12.4037 10.5262 13.6756 9.99938 15.0019 9.99938Z"
            fill="#F2F2F2"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root makeStyles-linkFont-11"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          fill="none"
          height="30"
          viewBox="0 0 30 30"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.42375 28.1248C20.745 28.1248 26.9381 18.7442 26.9381 10.6236C26.9381 10.3611 26.9381 10.0948 26.9269 9.83233C28.1326 8.9595 29.1732 7.87887 30 6.64108C28.8737 7.13768 27.6802 7.46497 26.4581 7.61233C27.7454 6.84291 28.7095 5.63228 29.1712 4.20546C27.9619 4.92194 26.6382 5.42491 25.2581 5.69233C24.3303 4.70425 23.1026 4.04968 21.7652 3.83001C20.4277 3.61033 19.0551 3.83781 17.86 4.47722C16.6649 5.11662 15.714 6.13226 15.1546 7.36682C14.5952 8.60138 14.4585 9.98596 14.7656 11.3061C12.3183 11.1834 9.92403 10.5476 7.7382 9.43998C5.55238 8.33237 3.62378 6.77767 2.0775 4.87671C1.29251 6.23249 1.05295 7.83621 1.40746 9.36221C1.76197 10.8882 2.68398 12.2221 3.98625 13.093C3.01039 13.0598 2.05598 12.7978 1.2 12.328V12.4123C1.20168 13.8326 1.6937 15.2087 2.59289 16.3081C3.49208 17.4075 4.74328 18.1627 6.135 18.4461C5.60674 18.5916 5.06106 18.6642 4.51312 18.6617C4.12684 18.6629 3.74134 18.6271 3.36188 18.5548C3.75522 19.7774 4.52114 20.8464 5.55234 21.612C6.58354 22.3776 7.82838 22.8014 9.1125 22.8242C6.93104 24.5376 4.23638 25.4668 1.4625 25.4623C0.973726 25.4644 0.485293 25.4362 0 25.378C2.81534 27.1728 6.08493 28.1259 9.42375 28.1248Z"
            fill="#F2F2F2"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root makeStyles-linkFont-11"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          
        </span>
      </div>
    </li>
  </ul>,
  <div
    className="makeStyles-buttonPos-10"
  >
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-7 MuiButton-contained makeStyles-button-9 MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiTypography-root MuiTypography-button"
        >
          Edytuj
        </span>
      </span>
    </button>
  </div>,
]
`;

exports[`UserInterests matches snapshot 1`] = `
Array [
  <ul
    className="MuiList-root makeStyles-root-12 MuiList-padding"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          
        </span>
      </div>
    </li>
  </ul>,
  <div
    className="makeStyles-buttonPos-14"
  >
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-7 MuiButton-contained makeStyles-button-13 MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiTypography-root MuiTypography-button"
        >
          Edytuj
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;
